### DISCUSSION TITLE
Merging with rsheftel/pandas_market_calendars?

### DISCUSSION THREAD

**Discussion by th-alanreed** [gerrymanoim/exchange_calendars#297] [General]
In https://github.com/quantopian/trading_calendars it says that this package and https://github.com/rsheftel/pandas_market_calendars are working to merge. Is that effort still underway? 

This package seems to have more active development but rsheftel/pandas_market_calendars has more calendars covered.

**Comment by maread99** [comment]
Hi @aarreedd

Thanks for raising the discussion.

`pandas_market_calendars` provides all the calendars defined in `exchange_calendars` (by mirroring them) plus some others that are native to `pandas_market_calendars`, hence pmc has more calendars. There's currently no intention to further merge the packages.

As for which to use, `pandas_market_calendars` is more lightweight than `exchange_calendars` although that comes with the loss of some of the `exchange_calendars` methods. On the other hand, I think some of pmc's native calendars offers additional market hours info concerning the likes of auction periods.

**Reply by th-alanreed** [reply]
Thanks for the explanation. It seems like `exchange_calendar` is the way to go for us.

I'm the founder of TradingHours.com - we provide paid trading hours and market holidays data for over 900 markets and have a dedicated team of researchers who keep the data up-to-date.

We're thinking about starting to distribute our data in a format compatible with `exchange_calendar`. It would still be paid, but I think this would make it easier for our customers to use the data.

I'm curious to hear your thoughts on whether you think this effort would be welcomed by the users and maintainers of this project.

If we go forward with this, we'd want to support the project in some way by contributing fixes or features, contributing updates to the free calendars, or by directly sponsoring the package.

**Reply by maread99** [reply]
Where things stand at the moment with regards to maintenance and development:
* All calendars are user-maintained. PRs updating existing calendars and defining new ones are always very much welcome (links to workflows can be found via the README).
* PRs offering fixes are also very much appreciated.
* Personally, I'm happy to review and oversee the implementation of development proposals that are of **general use** (a good example is #291). My input for such development is focused on quality control (ensuring the consistency of the public interface, that there are adequate tests, documentation etc).
* Any more esoteric requirements that are not of general interest are, as it stands, expected to be implemented by clients locally.

 @gerrymanoim is the contact if you're after anything outside the above and on the subject of sponsorship.

Cheers

**Reply by th-alanreed** [reply]
Thanks, @maread99 . I'll see about having our team submit PRs for the markets you cover when we see updates. 

@gerrymanoim - Can you reach out directly to discuss possible sponsorship? alan [at] tradinghours.com
